### PR TITLE
Create new shopping list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items
## Summary 
A shopping list is a set of items associated with a specific three-word token. We need to allow users to start new lists, so they can save the things they need to buy. To start a new list, we will need to do a few things:

1. Generate a new, unique token
2. Save the token to `localStorage`
3. Show the user the list view

Up to this point, the app has rendered the items in the list associated with the token `my test list`. Before we can allow users to make *new* lists, we have to stop showing them this test list.

## Acceptance criteria
- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] They are redirected from the List (`/list`) page back to the home (`/`) page.
- [x] A button in the `Home` component allows them to create a new list.
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the List (`/list`) page.

## Notes
- The  `generateToken` function from the `shopping-list-utils` package should be used to generate the token. More information on that package [its documentation](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils). Importing it will look something like this:
``` js
import { generateToken } from '@the-collab-lab/shopping-list-utils';
```
- The `setListToken` function takes **one** argument: the new list token.
- Eventually the welcome/home page will look like [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png). It’s good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
- For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by tialaaa**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

This PR adds functionality to allow users to create a new shopping list. We installed the `Shopping List Utils` package, and imported the `generateToken()` function to randomly generate a new list token. The `newToken` is saved into state and local storage using the `setListToken` function. If there is no token in local storage, the user will be redirected to the Home page. When the user clicks the `create a new list` button, a token will be generated and then saved in storage and the Firestore database, and the user is then redirected to the List page.

## Related Issue

Closes #3 

## Acceptance Criteria

- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] They are redirected from the List (`/list`) page back to the home (`/`) page.
- [x] A button in the `Home` component allows them to create a new list.
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the List (`/list`) page.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓   | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

Screenshot 2023-08-14 at 2 25 49 PM

### After

Screenshot 2023-08-21 at 4 17 43 PM

## Testing Steps / QA Criteria

- Before clicking any buttons, navigate to the `/list` path. You should automatically be redirected to the Home `/` page.
- From the Home page, click the `Create a new list` button. You should be redirected to the List `/list` page. The user's `listToken` auto-saves to the Firestore database while also saving to State and Local Storage. The List page will show the shopping items associated with the current list.

### CODE REVIEW

**Comment by Satoshi-Sh** [comment]
I have a question about this criteria.
If a user does already have a token:
 They are automatically redirected to the List (/list) page.

I think it's not working.
redirection


Except for this, it looks great.

**Review by Satoshi-Sh — APPROVED** [review]
All good!!

**Comment by polly89** [comment]
The code looks great. Really clean :) The interface works as expected. When I cleared the token it sent me to the home page. When I clicked "create list" it took me to the list page. 
Screenshot 2023-08-22 at 8 01 27 PM

**Review by polly89 — APPROVED** [review]
Nice work! Everything looks great.

File: src/api/firebase.js [hunk 1]
```diff
@@ -80,3 +80,14 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export async function createNewList(listId) {
+	try {
+		await addDoc(collection(db, listId), {});
+		console.log('Grocery list created for token: ', listId);
```
Comment (deeheber) [inline src/api/firebase.js]: **Suggestion:** You may want to consider removing this log statement. For now it's fine, but longterm we don't want that in our app.

File: src/App.jsx [hunk 2]
```diff
@@ -29,12 +30,33 @@ export function App() {
 	 */
 	const data = useShoppingListData(listToken);
 
+	function createToken() {
+		try {
+			const newToken = generateToken();
+			return newToken;
+		} catch {
+			console.log('Error: An error occurred while setting the new token');
```
Comment (deeheber) [inline src/App.jsx]: **Suggestion/Nit:** You might want to consider changing this to `console.error` for consistency with the other `catch` statement.

**Review by deeheber — APPROVED** [review]
Great job! All of the requirements have been met, and I like your communication style in the PR. ⭐️

Just made two super tiny suggestions that I'll let you two decide if you want to take. I'm fine with merging as is though, so ✅ with or without those suggestions.

File: src/api/firebase.js (hunk 1 above)
Comment (ismarjiw) [reply src/api/firebase.js]: Cleaned up :)

File: src/App.jsx (hunk 2 above)
Comment (ismarjiw) [reply src/App.jsx]: Replaced :)

**Pull request merged**